lib/command_graph.c

```diff
@@ -0,0 +1,480 @@
+/*
```

Comment:
should probably reformat the copyright header to use the style in the style guide while we're in here, and maybe kernel-style it as well